harp/reader.py

```diff
@@ -44,13 +44,53 @@ def __init__(
         self.read = read
 
 
+class RegisterMap:
+    registers: dict[str, RegisterReader]
+    _register_map: dict[int, str]
+
+    def __init__(self, registers: dict[str, RegisterReader]) -> None:
+        self.registers = registers
+        self._register_map = {self.registers[key].register.address: key for key in self.registers.keys()}
+
+    def __getitem__(self, __key: Union[str, int]) -> RegisterReader:
+        if isinstance(__key, int):
+            return self.registers[self._register_map[__key]]
+        elif isinstance(__key, str):
+            return self.registers[__key]
+        else:
+            raise TypeError(f"key must be int or str, not {type(__key)}")
+
+    def __iter__(self) -> Iterable[str]:
+        return iter(self.registers)
+
+    def __len__(self) -> int:
+        return len(self.registers)
+
+    def __repr__(self) -> str:
+        return repr(self.registers)
+
+    def __str__(self) -> str:
+        return str(self.registers)
+
+    def __dir__(self) -> Iterable[str]:
```

Comment:
We probably shouldn't override `__dir__` and leave that for the parent reader object.

Reply:
Removed